tests/api/utils/singletons/test_k8s_utils.py

```diff
@@ -58,8 +68,68 @@ def test_get_logger_pods_label_selector(
     if extra_selector:
         selector += f",{extra_selector}"
 
-    k8s_helper = server.api.utils.singletons.k8s.K8sHelper(namespace, silent=True)
     k8s_helper.list_pods = unittest.mock.MagicMock()
 
     k8s_helper.get_logger_pods(project, uid, run_type)
-    k8s_helper.list_pods.assert_called_once_with(namespace, selector=selector)
+    k8s_helper.list_pods.assert_called_once_with(
+        k8s_helper.namespace, selector=selector
+    )
+
+
+@pytest.mark.parametrize(
+    "k8s_secret_data,secrets_data",
+    [
+        ({"key1": "value1", "key2": "value2"}, []),
+        ({"key1": "value1", "key2": "value2"}, None),
+        (None, ["key1"]),
+        ({}, ["key1"]),
+    ],
+)
+def test_delete_secrets_no_changes(k8s_helper, k8s_secret_data, secrets_data):
+    k8s_helper.v1api.read_namespaced_secret.return_value = unittest.mock.MagicMock(
+        data=k8s_secret_data
+    )
+
+    result = k8s_helper.delete_secrets("my-secret", secrets_data)
+
+    assert result is None
+    k8s_helper.v1api.read_namespaced_secret.assert_called_once_with(
+        "my-secret", k8s_helper.namespace
+    )
+    k8s_helper.v1api.replace_namespaced_secret.assert_not_called()
+    k8s_helper.v1api.delete_namespaced_secret.assert_not_called()
+
+
+def test_delete_secrets_secret_found_with_changes(k8s_helper):
+    secret_data = {"key1": "value1", "key2": "value2"}
+    k8s_secret_mock = unittest.mock.MagicMock(data=secret_data)
+    k8s_helper.v1api.read_namespaced_secret.return_value = k8s_secret_mock
+
+    result = k8s_helper.delete_secrets("my-secret", ["key1"])
+
+    assert result == mlrun.common.schemas.SecretEventActions.updated
+    k8s_helper.v1api.read_namespaced_secret.assert_called_once_with(
+        "my-secret", k8s_helper.namespace
+    )
+    k8s_secret_mock.data = {"key2": "value2"}
+    k8s_helper.v1api.replace_namespaced_secret.assert_called_once_with(
+        "my-secret", k8s_helper.namespace, k8s_secret_mock
+    )
+    k8s_helper.v1api.delete_namespaced_secret.assert_not_called()
+
+
+def test_delete_secrets_delete_secret(k8s_helper):
```

Comment:
```suggestion
def test_ensure_secret_deletion_with_delete_secrets(k8s_helper):
```

Reply:
@yaelgen For the same reason below - how about that? 
`test_delete_secrets_confirms_secret_deletion_when_keys_match`